### ISSUE TITLE
Unable to enable cluster mode

### ISSUE THREAD

**Issue by rymancl** [cloudposse/terraform-aws-elasticache-redis#212]
### Describe the Bug

I am trying to switch an existing cluster with `cluster_mode_enabled = false` (default) to `cluster_mode_enabled = true`.
As part of this change, I have also made the following changes:
* remove `cluster_size`
* set `automatic_failover_enabled = true`
* set `cluster_mode_num_node_groups = 1`
* set `cluster_mode_replicas_per_node_group = 1`

Proposed plan:
![Screenshot 2023-12-12 at 12 22 56 PM](https://github.com/cloudposse/terraform-aws-elasticache-redis/assets/3211227/80686be1-026b-4e04-bf80-0985d2984e85)

Apply error:
![Screenshot 2023-12-12 at 12 26 38 PM](https://github.com/cloudposse/terraform-aws-elasticache-redis/assets/3211227/cde83f3a-ace8-4cd4-8d7a-f90c17e2bb96)

```
module.shared-redis.aws_elasticache_parameter_group.default[0]: Modifying... [id=<redacted>-uw2-shared]
╷
│ Error: modifying ElastiCache Parameter Group: InvalidParameterValue: The parameter cluster-enabled cannot be modified.
│       status code: 400, request id: <redacted>
│ 
│   with module.shared-redis.aws_elasticache_parameter_group.default[0],
│   on .terraform/modules/shared-redis/main.tf line 91, in resource "aws_elasticache_parameter_group" "default":
│   91: resource "aws_elasticache_parameter_group" "default" {
│ 
╵
```

According to AWS, this parameter cannot be modified:
![Screenshot 2023-12-12 at 12 28 14 PM](https://github.com/cloudposse/terraform-aws-elasticache-redis/assets/3211227/0a920ef1-1700-4db8-8188-bcf9801f3684)

### Expected Behavior

Successfully enable cluster mode on Redis 

### Steps to Reproduce

See above

### Environment

* OSX Sonoma 14.2
* Terraform v1.6.5 on darwin_arm64
* AWS provider version 5.30.0
* Module version 0.53.0

**Comment by kevcube** [comment]
@rymancl it looks like AWS disallows changing of this parameter. I suspect the only way to do this is to delete the cluster and restore from a snapshot with cluster mode enabled.

**Comment by kevcube** [comment]
@rymancl or at least you can not manually change it. Maybe if you add another instance to the "cluster" then AWS will change that parameter to true behind the scenes.

**Comment by rymancl** [comment]
> @rymancl it looks like AWS disallows changing of this parameter. I suspect the only way to do this is to delete the cluster and restore from a snapshot with cluster mode enabled.

That's true, but as of redis 7.0 AWS allows you to do in place moves from cluster mode disabled to cluster mode enabled. 

Docs [here](https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/VersionManagement.html#VersionManagement-upgrade-considerations)

> Beginning with Redis 7, ElastiCache for Redis supports switching between Redis (cluster mode disabled) and Redis (cluster mode enabled).

The bug I believe is that the module is trying to set that parameter when it shouldn't be. I assume AWS will set it for you.

**Comment by kevcube** [comment]
Ah I see. Try a targeted apply (assuming these are dev/sandbox resources) to `module.shared-redis.aws_elasticache_replication_group.default[0]` then see if AWS changes that parameter. Terraform should pick it up before the next plan and won't attempt to change it again.

Maybe the ideal solution would be for this module to not set it at all and assume that AWS manages it, depending on the outcome of the above.
